src/views/open-api-docs-view/components/open-api-sidebar-contents/utils/filter-by-title.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Given a flat array of items, and a `filterValue`,*
+ * Return an array of filtered items, including only those items that
+ * are objects with a `title` attribute that matches the `filterValue`.
+ *
+ * If the provided `filterValue` is falsy, we return all items unchanged.
+ */
+export function filterByTitle<T = unknown>(
+	items: T[],
```

Comment:
Should we just be explicit with the type here that we expect `title`? Or do you prefer the type guard? Curious if we need to give any feedback if there isn't a title on the items for DX.

Reply:
I'm using the vague type & type guard cause we do expect _some_ items to not have a title... this is a great point on DX though 🤔 I'll update to log a lil warning in dev if _none_ of the incoming items have a `title` property!